Makefile.common

```diff
@@ -33,10 +35,15 @@ endif
 
 # Check that rustc is the right nightly - warn not error if wrong, sometimes useful to test others
 RUSTC_VERSION_STRING := rustc 1.16.0-nightly (83c2d9523 2017-01-24)
-ifneq ($(shell rustc --version),$(RUSTC_VERSION_STRING))
-    $(warning Unexpected rustc version. Expected $(RUSTC_VERSION_STRING) got $(shell rustc --version))
+ifeq ($(shell rustup -V &> /dev/null && echo success),success)
+  DUMMY := $(shell rustup install $(RUSTUP_TOOLCHAIN))
```

Comment:
Like the new language

It's a bit hacky, but the solution that jumps to mind is to just nest the checks:

```
ifeq ($(shell rustup -V &> /dev/null && echo success),success)
    DUMMY := $(shell rustup install $(RUSTUP_TOOLCHAIN))
    ifneq ($(shell RUST_TOOLCHAIN=$(RUSTUP_TOOLCHAIN) rustc --version), $(RUSTC_VERSION_STRING))
        $(error rustup failed. SAD)
```